packages/webapp/src/delegates/api/eden-contract.ts

```
@@ -1,31 +1,92 @@
+const MEMBER_REPRESENTATIVE_IF_NOT_PARTICIPATED_IN_RECENT_ELECTION =
+    "zzzzzzzzzzzzj";
+const MEMBER_REPRESENTATIVE_IF_FAILED_TO_REACH_CONSENSUS = "";
+const memberHasRepresentative = (member: EdenMember) =>
+    member.account !== MEMBER_REPRESENTATIVE_IF_FAILED_TO_REACH_CONSENSUS &&
+    member.account !==
+        MEMBER_REPRESENTATIVE_IF_NOT_PARTICIPATED_IN_RECENT_ELECTION;
 
-const hasMemberRSVPed = () => {
-    return false; // TODO
-};
+export const getMyDelegation = async (
+    loggedInMemberAccount: string | undefined
+): Promise<EdenMember[]> => {
+    let myDelegates: EdenMember[] = [];
 
-export const getMyDelegation = () => {
-    return {}; // TODO
+    if (!loggedInMemberAccount) return myDelegates;
+
+    try {
+        const leadRepresentative = await queryClient.fetchQuery(
+            queryHeadDelegate.queryKey,
+            queryHeadDelegate.queryFn
+        );
+
+        // NEXT: add fixtures for each of these querys and test the logic (that I couldn't test before)
+
+        const { queryKey, queryFn } = queryMemberByAccountName(
+            loggedInMemberAccount
+        );
+        let nextDelegate: EdenMember = await queryClient.fetchQuery(
+            queryKey,
+            queryFn
+        );
+        if (!nextDelegate || !leadRepresentative) return myDelegates;
+
+        while (
+            nextDelegate!.account !== leadRepresentative &&
+            memberHasRepresentative(nextDelegate)
+        ) {
+            myDelegates.push(nextDelegate);
+            const { queryKey, queryFn } = queryMemberByAccountName(
+                nextDelegate!.representative
+            );
+            nextDelegate = await queryClient.fetchQuery(queryKey, queryFn);
+            if (!nextDelegate) return myDelegates;
+        }
+        if (
+            nextDelegate.account === leadRepresentative &&
+            memberHasRepresentative(nextDelegate)
+        ) {
+            myDelegates.push(nextDelegate);
+        }
+    } catch (e) {
```

I know we talked about this too...but as I think about it more, the worst that happens here if we just let this throw is that we get no data at the query/UI level and an error, right?

---

true